### PR TITLE
Next copy screen

### PR DESCRIPTION
**Pull request by GantMan**
the ability to copy screen examples... untested due to my local environment, but with a LOT of copy and paste and micro-testing I'm sure this works.  So here's the PR.

This will copy listview examples to the plugin screen.

### CODE REVIEW

File: packages/ignite/src/extensions/ignite.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
-const { dissoc, merge, pipe, prop, sortBy, propSatisfies, filter } = require('ramda')
+// Steeeeeeeve I'm starting to not love this giant deconstruct - leaving it here to prove why
+const { dissoc, merge, pipe, prop, sortBy, propSatisfies, filter, reduce, flatten, map, takeLast, split, replace } = require('ramda')
```
Comment (skellock) [inline packages/ignite/src/extensions/ignite.js]: We chatted about this offline.  I'll refactor this file right after this lands.

File: packages/ignite/src/extensions/ignite.js [hunk 2]
```diff
@@ -181,6 +183,157 @@ function attach (plugin, command, context) {
...
+   *   {title: 'Row Example', screen: 'Row.js', ancillary: ['file1', 'file2']},
+   *   {title: 'Grid Example', screen: 'Grid.js', ancillary: ['file']},
+   *   {title: 'Section Example', screen: 'Section.js', ancillary: ['file']},
+   * ])
+   */
+  async function addScreenExamples (files, props = {}) {
+    const { filesystem, patching, ignite } = context
+    const config = ignite.loadIgniteConfig()
+    // consider this being part of context.ignite
+    const pluginName = takeLast(1, split(path.sep, ignitePluginPath()))[0]
```
Comment (skellock) [inline packages/ignite/src/extensions/ignite.js]: Roger.

File: packages/ignite/src/extensions/ignite.js [hunk 3]
```diff
@@ -181,6 +183,157 @@ function attach (plugin, command, context) {
...
+
+    // currently only supporting 1 form of examples
+    if (config.examples === 'classic') {
+      const spinner = print.spin(`▸ adding screen examples`)
+
+      // merge and flatten all dem files yo.
+      let allFiles = reduce((acc, v) => {
```
Comment (skellock) [inline packages/ignite/src/extensions/ignite.js]:
```js
pipe(
  map(props(['screen', 'ancillary'])),
  flatten,
  reject(isNil),
  uniq
)(data)
```

I swear to god they should have called that library Pipeda.

File: packages/ignite-basic-generators/index.js [hunk 4]
```diff
@@ -1,9 +1,27 @@
+const screenExamples = [
+  {
+    title: 'Row Example', 
+    screen: 'examples/RowExample.js', 
+    ancillary: ['examples/Styles/RowExampleStyle.js']
```
Comment (skellock) [inline packages/ignite-basic-generators/index.js]: I find the word `ancillary` confusing.  Your vocabulary is much richer than mine though.  I'd have chosen `files` and would have explained in the docs these are extra goodies.

File: packages/ignite-animatable/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ignite-animatable",
-  "version": "0.2.0-alpha.2",
+  "version": "0.2.0-alpha.11",
```
Comment (skellock) [inline packages/ignite-animatable/package.json]: 9 rounds of pain.  I feel for ya dude.

File: packages/ignite-basic-generators/templates/examples/GridExample.js.ejs [hunk 6]
```diff
@@ -0,0 +1,131 @@
+// @flow
+
+import React, { PropTypes } from 'react'
+import { View, Text, ListView } from 'react-native'
+import { connect } from 'react-redux'
+// import { Actions as NavigationActions } from 'react-native-router-flux'
```
Comment (skellock) [inline packages/ignite-basic-generators/templates/examples/GridExample.js.ejs]: A lot of noise & garbage coming in here with these examples.  Let's clean these up in the future.

File: packages/ignite-basic-generators/templates/examples/GridExample.js.ejs [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+class GridExample extends React.Component {
+
+  state: {
+    dataSource: Object
+  }
+
+  constructor (props) {
+    super(props)
+    // If you need scroll to bottom, consider http://bit.ly/2bMQ2BZ
```
Comment (skellock) [inline packages/ignite-basic-generators/templates/examples/GridExample.js.ejs]: React Native supports this now.

File: packages/ignite-basic-generators/templates/examples/GridExample.js.ejs [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+  /* ***********************************************************
+  * STEP 3
+  * `_renderRow` function -How each cell/row should be rendered
+  * It's our best practice to place a single component here:
+  *
+  * e.g.
+    return <MyCustomCell title={rowData.title} description={rowData.description} />
+  *************************************************************/
+  _renderRow (rowData) {
```
Comment (skellock) [inline packages/ignite-basic-generators/templates/examples/GridExample.js.ejs]: Not a fan of underscore functions.  I know these are old example files you're bringing in though.

File: packages/ignite-basic-generators/templates/listview-grid-style.ejs [hunk 9]
```diff
@@ -1,7 +1,7 @@
 // @flow
 
 import { StyleSheet } from 'react-native'
-import { ApplicationStyles, Metrics, Colors } from '../../Themes/'
+import { ApplicationStyles, Metrics, Colors } from '../../../../../DevScreens/DevTheme/'
```
Comment (skellock) [inline packages/ignite-basic-generators/templates/listview-grid-style.ejs]: OK.  I'm onboard for that babel root path plugin btw.  Every time I see this now, I cringe.

File: packages/ignite-dev-screens/templates/PluginExamplesScreen.js [hunk 10]
```diff
@@ -1,14 +1,19 @@
 // @flow
 
-import React, { PropTypes } from 'react'
+// Fair Warning:  PluginExamples has a good bit of Ignite automation in editing.
```
Comment (skellock) [inline packages/ignite-dev-screens/templates/PluginExamplesScreen.js]: I'm not sure this comment needs to be here.  I think this comment rightfully applies to every single template we hand off to the end-user.

File: packages/ignite-dev-screens/templates/PluginExamplesScreen.js [hunk 11]
```diff
@@ -18,7 +23,7 @@ class PluginExamplesScreen extends React.Component {
     return (
       <View style={styles.mainContainer}>
         <Image source={Images.background} style={styles.backgroundImage} resizeMode='stretch' />
-        <TouchableOpacity onPress={()=> this.props.navigation.goBack()} style={{
+        <TouchableOpacity onPress={() => this.props.navigation.goBack(null)} style={{
```
Comment (skellock) [inline packages/ignite-dev-screens/templates/PluginExamplesScreen.js]: That's a bummer they force you to put `null` eh?

File: packages/ignite-basic-generators/index.js (hunk 4 above)
Comment (jamonholmgren) [inline packages/ignite-basic-generators/index.js]: I agree. I'd prefer `files` or `extraFiles`. `ancillary` will confuse non-native English speakers too.

File: packages/ignite-basic-generators/templates/listview-grid-style.ejs (hunk 9 above)
Comment (jamonholmgren) [inline packages/ignite-basic-generators/templates/listview-grid-style.ejs]: `../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../IAgree.js`

**Pull request merged**